tests/testthat/test-thumbnail.R

```diff
@@ -0,0 +1,120 @@
+mock_dirs <- c(
+  "unversioned" = "2024.08.0",
+  "v1" = "2024.09.0"
+)
+
```

Comment:
This is a more or less stylistic suggestion, but what if you used the version numbers as the only thing that switched here? As in, don't name them (you can and should still comment and explain that that is where the transition for thumbnails happened, of course). But then you don't need to juggle both names and versions in the test versions down below. 

Thinking about our future selves: when I see the error about `get_thumbnail() gets the thumbnail v1` I'm not sure I would know that I need to go to the top of this file to find the mapping (or that v1 for thumbnails means I'm in the 2024.09.0 directory). But if it were `get_thumbnail() gets the thumbnail 2024.09.0` I have some idea more directly that I should look there.

Reply:
I see your point — how would you feel about having both in the test name?

Reply:
(I put both in the name)